e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/filter-payment-details.spec.js

```diff
@@ -0,0 +1,76 @@
+import {
+  FeeRecordEntityMockBuilder,
+  PaymentEntityMockBuilder,
+  UTILISATION_REPORT_RECONCILIATION_STATUS,
+  UtilisationReportEntityMockBuilder,
+} from '@ukef/dtfs2-common';
+import pages from '../../pages';
+import { NODE_TASKS } from '../../../../../e2e-fixtures';
+import USERS from '../../../fixtures/users';
+
+context(`users can filter payment details by facility id and payment reference and currency`, () => {
+  const bankId = '961';
+  const reportId = 12;
+
+  const utilisationReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS)
+    .withId(reportId)
+    .withBankId(bankId)
+    .withDateUploaded(new Date())
+    .build();
+
+  before(() => {
+    cy.task(NODE_TASKS.REINSERT_ZERO_THRESHOLD_PAYMENT_MATCHING_TOLERANCES);
+  });
+
+  beforeEach(() => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [utilisationReport]);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`/utilisation-reports/${reportId}`);
```

Comment:
Not a blocker as can see a few instances of this - we should mimic an actual user journey rather than navigating to a URL directly

Reply:
I see your point about wanting to truly test the end-to-end journey, I'm not sure how practical this is though (see below):

Instead of `cy.visit` we could try to do something like:

```
pages.utilisationReportsSummaryPage.visit();
pages.utilisationReportsSummaryPage.reportLink(BANK_ID, SUBMISSION_MONTH).click();
```

We have `BANK_ID`, but `SUBMISSION_MONTH` is a bit trickier to construct/deal with - [this is done in `e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/reconcile-fee-records.spec.js`](https://github.com/UK-Export-Finance/dtfs2/blob/9159cd13c17e74be99d3ea03b159eb9677e137b4/e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/reconcile-fee-records.spec.js#L55) - and requires a few calls to different helpers.

Is this the sort of approach you were thinking? I can add a similar implementation here, it just seems a bit more overhead/adding more complexity to the tests

Reply:
Thanks @sw-mattw I know it adds some complexity but `.visit()` doesn't replicate "typical / happy path" kind of user experience, we could have something like `cy.goToReport(bankId)` for example and use throughout various tests, so it's all handled in one place and reduces such lines in individual tests.

I think this should be done in one PR and for many other things; Happy to resolve since there are other instances of this unrelated to this PR and don't want them to have different approaches.

Reply:
@ttbarnes I've added a ticket to address this thanks - FN-3483